File: MAP/Lab6/src/Model/Statement/CompStmt.java

package Model.Statement;

public class CompStmt implements Statement
{
    private Statement first, second;
    public CompStmt(Statement f, Statement s)
    {
        first = f;
        second = s;
    }

    @Override
    public PrgState execute(PrgState state)
    {
        state.getExecStack().push(second);
        state.getExecStack().push(first);
        return null;
    }

    @Override
    public String toString()
    {
        return "" + first + second;
    }
}

File: MAP/LAB7-MAP/LAB7-MAP/Statements/CompStmt.cs
namespace MAP_Lab7.Statements
{
    public class CompStmt:Statement
    {
        private Statement first, second;
        public CompStmt(Statement f, Statement s)
        {
            first = f;
            second = s;
        }

        public PrgState Execute(PrgState state)
        {
            state.ExeStack.PushS(second);
            state.ExeStack.PushS(first);
            return state;
        }

        public override string ToString()
        {
            return "" + first + second;
        }
    }
}
